plugins/flutter_aepcore/ios/Classes/FlutterAEPCorePlugin.m

```diff
@@ -112,7 +112,20 @@ - (void)handleDispatchEvent:(FlutterMethodCall *)call
 
 - (void)handleDispatchEventWithResponseCallback:(FlutterMethodCall *)call
                                             result:(FlutterResult)result {
-    NSDictionary *eventDict = (NSDictionary *)call.arguments;
+    NSDictionary *eventDict = (NSDictionary *)call.arguments[@"eventData"];
+    NSNumber *timeoutNumber = call.arguments[@"timeout"];
+    double timeout = 1;
+    if ([timeoutNumber respondsToSelector:@selector(intValue)]) {
```

Comment:
doubleValue

Reply:
We do not need to convert the timeout value to a double type in this case because the dart API accepts it as an Integer.

Reply:
Oh, I see the issue.  The code flow should convert the timtouNuber variable to an Integer.